tile_map not compiling on Saucy

It looks like a new YAML API is being used. 

http://build.ros.org/view/Ibin_uS64/job/Ibin_uS64__tile_map__ubuntu_saucy_amd64__binary/14/

```
00:44:49 [ 94%] Building CXX object CMakeFiles/tile_map_plugin.dir/src/tile_map_plugin.cpp.o
00:44:49 /usr/bin/c++   -DQT_CORE_LIB -DQT_GUI_LIB -DQT_NETWORK_LIB -DQT_NO_DEBUG -DQT_NO_KEYWORDS -DQT_OPENGL_LIB -DROSCONSOLE_BACKEND_LOG4CXX -DROS_PACKAGE_NAME=\"tile_map\" -Dtile_map_plugin_EXPORTS -g -O2 -fstack-protector --param=ssp-buffer-size=4 -Wformat -Werror=format-security -DNDEBUG -D_FORTIFY_SOURCE=2  -O2 -g -DNDEBUG -fPIC -isystem /usr/include/qt4 -isystem /usr/include/qt4/QtOpenGL -isystem /usr/include/qt4/QtGui -isystem /usr/include/qt4/QtNetwork -isystem /usr/include/qt4/QtCore -I/tmp/binarydeb/ros-indigo-tile-map-0.0.6/include -I/opt/ros/indigo/include -I/usr/include/jsoncpp -I/tmp/binarydeb/ros-indigo-tile-map-0.0.6/obj-x86_64-linux-gnu -I/tmp/binarydeb/ros-indigo-tile-map-0.0.6    -o CMakeFiles/tile_map_plugin.dir/src/tile_map_plugin.cpp.o -c /tmp/binarydeb/ros-indigo-tile-map-0.0.6/src/tile_map_plugin.cpp
00:44:57 /tmp/binarydeb/ros-indigo-tile-map-0.0.6/src/tile_map_plugin.cpp: In member function ‘virtual void tile_map::TileMapPlugin::LoadConfig(const YAML::Node&, const string&)’:
00:44:57 /tmp/binarydeb/ros-indigo-tile-map-0.0.6/src/tile_map_plugin.cpp:337:7: error: ‘const_iterator’ is not a member of ‘YAML::Node’
00:44:57        YAML::Node::const_iterator source_iter;
00:44:57        ^
00:44:57 /tmp/binarydeb/ros-indigo-tile-map-0.0.6/src/tile_map_plugin.cpp:337:34: error: expected ‘;’ before ‘source_iter’
00:44:57        YAML::Node::const_iterator source_iter;
00:44:57                                   ^
00:44:57 /tmp/binarydeb/ros-indigo-tile-map-0.0.6/src/tile_map_plugin.cpp:338:12: error: ‘source_iter’ was not declared in this scope
00:44:57        for (source_iter = sources.begin(); source_iter != sources.end(); source_iter++)
00:44:57             ^
00:44:57 /tmp/binarydeb/ros-indigo-tile-map-0.0.6/src/tile_map_plugin.cpp:344:59: error: expected primary-expression before ‘>’ token
00:44:57            type = ((*source_iter)[TYPE_KEY]).as<std::string>();
00:44:57                                                            ^
00:44:57 /tmp/binarydeb/ros-indigo-tile-map-0.0.6/src/tile_map_plugin.cpp:344:61: error: expected primary-expression before ‘)’ token
00:44:57            type = ((*source_iter)[TYPE_KEY]).as<std::string>();
00:44:57                                                              ^
00:44:57 /tmp/binarydeb/ros-indigo-tile-map-0.0.6/src/tile_map_plugin.cpp:350:79: error: expected primary-expression before ‘>’ token
00:44:57                QString::fromStdString(((*source_iter)[NAME_KEY]).as<std::string>()),
00:44:57                                                                                ^
00:44:57 /tmp/binarydeb/ros-indigo-tile-map-0.0.6/src/tile_map_plugin.cpp:350:81: error: expected primary-expression before ‘)’ token
00:44:57                QString::fromStdString(((*source_iter)[NAME_KEY]).as<std::string>()),
00:44:57                                                                                  ^
00:44:57 /tmp/binarydeb/ros-indigo-tile-map-0.0.6/src/tile_map_plugin.cpp:351:81: error: expected primary-expression before ‘>’ token
00:44:57                QString::fromStdString((*source_iter)[BASE_URL_KEY].as<std::string>()),
00:44:57                                                                                  ^
00:44:57 /tmp/binarydeb/ros-indigo-tile-map-0.0.6/src/tile_map_plugin.cpp:351:83: error: expected primary-expression before ‘)’ token
00:44:57                QString::fromStdString((*source_iter)[BASE_URL_KEY].as<std::string>()),
00:44:57                                                                                    ^
00:44:57 /tmp/binarydeb/ros-indigo-tile-map-0.0.6/src/tile_map_plugin.cpp:353:47: error: expected primary-expression before ‘int’
00:44:57                (*source_iter)[MAX_ZOOM_KEY].as<int>());
00:44:57                                                ^
00:44:57 /tmp/binarydeb/ros-indigo-tile-map-0.0.6/src/tile_map_plugin.cpp:358:79: error: expected primary-expression before ‘>’ token
00:44:57                QString::fromStdString(((*source_iter)[NAME_KEY]).as<std::string>()));
00:44:57                                                                                ^
00:44:57 /tmp/binarydeb/ros-indigo-tile-map-0.0.6/src/tile_map_plugin.cpp:358:81: error: expected primary-expression before ‘)’ token
00:44:57                QString::fromStdString(((*source_iter)[NAME_KEY]).as<std::string>()));
00:44:57                                                                                  ^
00:44:57 /tmp/binarydeb/ros-indigo-tile-map-0.0.6/src/tile_map_plugin.cpp:365:27: error: could not convert ‘(& node)->YAML::Node::operator[]<std::basic_string<char> >((*(const std::basic_string<char>*)(& tile_map::TileMapPlugin::BING_API_KEY)))’ from ‘const YAML::Node’ to ‘bool’
00:44:57      if (node[BING_API_KEY])
00:44:57                            ^
00:44:57 /tmp/binarydeb/ros-indigo-tile-map-0.0.6/src/tile_map_plugin.cpp:367:44: error: ‘const class YAML::Node’ has no member named ‘as’
00:44:57        std::string key = node[BING_API_KEY].as<std::string>();
00:44:57                                             ^
00:44:57 /tmp/binarydeb/ros-indigo-tile-map-0.0.6/src/tile_map_plugin.cpp:367:58: error: expected primary-expression before ‘>’ token
00:44:57        std::string key = node[BING_API_KEY].as<std::string>();
00:44:57                                                           ^
00:44:57 /tmp/binarydeb/ros-indigo-tile-map-0.0.6/src/tile_map_plugin.cpp:367:60: error: expected primary-expression before ‘)’ token
00:44:57        std::string key = node[BING_API_KEY].as<std::string>();
00:44:57                                                             ^
00:44:57 /tmp/binarydeb/ros-indigo-tile-map-0.0.6/src/tile_map_plugin.cpp:372:25: error: could not convert ‘(& node)->YAML::Node::operator[]<std::basic_string<char> >((*(const std::basic_string<char>*)(& tile_map::TileMapPlugin::SOURCE_KEY)))’ from ‘const YAML::Node’ to ‘bool’
00:44:57      if (node[SOURCE_KEY])
00:44:57                          ^
00:44:57 /tmp/binarydeb/ros-indigo-tile-map-0.0.6/src/tile_map_plugin.cpp:374:45: error: ‘const class YAML::Node’ has no member named ‘as’
00:44:57        std::string source = node[SOURCE_KEY].as<std::string>();
00:44:57                                              ^
00:44:57 /tmp/binarydeb/ros-indigo-tile-map-0.0.6/src/tile_map_plugin.cpp:374:59: error: expected primary-expression before ‘>’ token
00:44:57        std::string source = node[SOURCE_KEY].as<std::string>();
00:44:57                                                            ^
00:44:57 /tmp/binarydeb/ros-indigo-tile-map-0.0.6/src/tile_map_plugin.cpp:374:61: error: expected primary-expression before ‘)’ token
00:44:57        std::string source = node[SOURCE_KEY].as<std::string>();
00:44:57                                                              ^
00:44:58 make[4]: *** [CMakeFiles/tile_map_plugin.dir/src/tile_map_plugin.cpp.o] Error 1
00:44:58 make[4]: Leaving directory `/tmp/binarydeb/ros-indigo-tile-map-0.0.6/obj-x86_64-linux-gnu'
00:44:58 make[3]: *** [CMakeFiles/tile_map_plugin.dir/all] Error 2
00:44:58 make[3]: Leaving directory `/tmp/binarydeb/ros-indigo-tile-map-0.0.6/obj-x86_64-linux-gnu'
00:44:58 make[2]: *** [all] Error 2
00:44:58 make[2]: Leaving directory `/tmp/binarydeb/ros-indigo-tile-map-0.0.6/obj-x86_64-linux-gnu'
00:44:58 dh_auto_build: make -j1 returned exit code 2
00:44:58 make[1]: *** [override_dh_auto_build] Error 2
00:44:58 make[1]: Leaving directory `/tmp/binarydeb/ros-indigo-tile-map-0.0.6'
00:44:58 make: *** [build] Error 2
00:44:58 dpkg-buildpackage: error: debian/rules build gave error exit status 2
```

--- a/tile_map/src/tile_map_plugin.cpp
+++ b/tile_map/src/tile_map_plugin.cpp
@@ -338,40 +338,49 @@ namespace tile_map
       for (source_iter = sources.begin(); source_iter != sources.end(); source_iter++)
       {
         std::string type = "";
-        if ((*source_iter)[TYPE_KEY])
+        if (swri_yaml_util::FindValue(*source_iter, TYPE_KEY))
         {
           // If the type isn't set, we'll assume it's WMTS
-          type = ((*source_iter)[TYPE_KEY]).as<std::string>();
+          (*source_iter)[TYPE_KEY] >> type;
         }
         boost::shared_ptr<TileSource> source;
         if (type == "wmts" || type.empty())
         {
+          std::string name;
+          std::string base_url;
+          int max_zoom;
+          (*source_iter)[NAME_KEY] >> name;
+          (*source_iter)[BASE_URL_KEY] >> base_url;
+          (*source_iter)[MAX_ZOOM_KEY] >> max_zoom;
           source = boost::make_shared<WmtsSource>(
-              QString::fromStdString(((*source_iter)[NAME_KEY]).as<std::string>()),
-              QString::fromStdString((*source_iter)[BASE_URL_KEY].as<std::string>()),
+              QString::fromStdString(name),
+              QString::fromStdString(base_url),
               true,
-              (*source_iter)[MAX_ZOOM_KEY].as<int>());
+              max_zoom);
         }
         else if (type == "bing")
         {
-          source = boost::make_shared<BingSource>(
-              QString::fromStdString(((*source_iter)[NAME_KEY]).as<std::string>()));
+          std::string name;
+          (*source_iter)[NAME_KEY] >> name;
+          source = boost::make_shared<BingSource>(QString::fromStdString(name));
         }
         tile_sources_[source->GetName()] = source;
         ui_.source_combo->addItem(source->GetName());
       }
     }
 
-    if (node[BING_API_KEY])
+    if (swri_yaml_util::FindValue(node, BING_API_KEY))
     {
-      std::string key = node[BING_API_KEY].as<std::string>();
+      std::string key;
+      node[BING_API_KEY] >> key;
       BingSource* source = static_cast<BingSource*>(tile_sources_[BING_NAME].get());
       source->SetApiKey(QString::fromStdString(key));
     }
     
-    if (node[SOURCE_KEY])
+    if (swri_yaml_util::FindValue(node, SOURCE_KEY))
     {
-      std::string source = node[SOURCE_KEY].as<std::string>();
+      std::string source;
+      node[SOURCE_KEY] >> source;
 
       int index = ui_.source_combo->findText(QString::fromStdString(source), Qt::MatchExactly);
 

